examples/visualization/plot_study.ipynb

```diff
@@ -0,0 +1,261 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Visualizing High-dimensional Parameter Relationships\n",
+    "\n",
+    "This notebook demonstrates various visualizations of studies in Optuna.\n",
+    "The hyperparameters of a neural network trained to classify images are optimized and the resulting study is then visualized using these features.\n",
+    "\n",
+    "**Note:** If a parameter contains missing values, a trial with missing values is not plotted."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Preparing the Dataset"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "from sklearn.datasets import fetch_openml\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "mnist = fetch_openml(name='Fashion-MNIST', version=1)\n",
+    "classes = list(set(mnist.target))\n",
+    "x_train, x_test, y_train, y_test = train_test_split(mnist.data, mnist.target)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Defining the Objective Function"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from sklearn.neural_network import MLPClassifier\n",
+    "\n",
+    "def objective(trial):\n",
+    "    layers = []\n",
+    "    n_layers = trial.suggest_int('n_layers', 1, 3)\n",
+    "    for i in range(n_layers):\n",
+    "        layers.append(trial.suggest_int('n_units_l{}'.format(i), 32, 64))\n",
+    "\n",
+    "    clf = MLPClassifier(hidden_layer_sizes=tuple(layers))\n",
+    "\n",
+    "    for step in range(100):\n",
+    "        clf.partial_fit(x_train, y_train, classes=classes)\n",
+    "\n",
+    "        intermediate_value = 1.0 - clf.score(x_test, y_test)  # Report intermediate objective value.\n",
+    "        trial.report(intermediate_value, step)\n",
+    "\n",
+    "        if trial.should_prune(step):\n",
+    "            raise optuna.exceptions.TrialPruned()  # Handle pruning based on the intermediate value.\n",
+    "\n",
+    "    return clf.score(x_test, y_test)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Running the Optimization"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import optuna\n",
+    "\n",
+    "optuna.logging.set_verbosity(optuna.logging.WARNING)  # This verbosity change is just to simplify the notebook output.\n",
+    "\n",
+    "study = optuna.create_study(direction='maximize', pruner=optuna.pruners.MedianPruner())\n",
+    "study.optimize(objective, n_trials=32)"
```

Comment:
It takes about 30 mins to execute this cell in Google Colab.
Can we reduce the execution time by changing the size of the dataset like other examples?

Reply:
Changed to only use a subsample of 4k images (in contrast to previous 70k) in https://github.com/optuna/optuna/pull/735/commits/bd019b2da01ac9931336214291ddf676aa512dfa. Took roughly 5 min on my macbook.